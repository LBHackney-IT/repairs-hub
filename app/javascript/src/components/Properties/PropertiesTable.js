import React from 'react'
import PropertyRow from '../Property/PropertyRow'

const PropertiesTable = ({ properties }) => (
  <div>
    <hr className="govuk-section-break govuk-section-break--l govuk-section-break--visible" />

    <p className="govuk-heading-s">We found {properties.length} matching results</p>

    <table className="govuk-table govuk-!-margin-top-5">
      <thead className="govuk-table__head">
        <tr className="govuk-table__row">
          <th scope="col" className="govuk-table__header">Address</th>
          <th scope="col" className="govuk-table__header">Postcode</th>
          <th scope="col" className="govuk-table__header">Property type</th>
          <th scope="col" className="govuk-table__header">Property reference</th>
        </tr>
      </thead>
      <tbody className="govuk-table__body">
        {properties.map((property, index) => (
          <PropertyRow key={index} {...property} />
        ))}
      </tbody>
    </table>
  </div>
)

export default PropertiesTable
