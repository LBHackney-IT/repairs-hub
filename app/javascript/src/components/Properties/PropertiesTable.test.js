import React from 'react'
import { render } from '@testing-library/react'
import PropertiesTable from './PropertiesTable'

describe('PropertiesTable component', () => {
  const props = {
    properties: [
      {
        "propertyReference": "00023404",
        "address": {
          "shortAddress": "16 Pitcairn House  St Thomass Square",
          "postalCode": "E9 6PT"
        },
        "hierarchyType": {
          "levelCode": "7",
          "subTypeCode": "DWE",
          "subTypeDescription": "Dwelling"
        }
      },
      {
        "propertyReference": "00023389",
        "address": {
          "shortAddress": "1 Pitcairn House  St Thomass Square",
          "postalCode": "E9 6PT"
        },
        "hierarchyType": {
          "levelCode": "7",
          "subTypeCode": "DWE",
          "subTypeDescription": "Dwelling"
        }
      }
    ],
  }
  it('should render properly', () => {
    const { asFragment } = render(<PropertiesTable {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
