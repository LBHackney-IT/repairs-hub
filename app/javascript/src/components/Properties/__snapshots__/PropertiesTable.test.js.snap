// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesTable component should render properly 1`] = `
<DocumentFragment>
  <div>
    <hr
      class="govuk-section-break govuk-section-break--l govuk-section-break--visible"
    />
    <table
      class="govuk-table"
    >
      <caption
        class="govuk-table__caption"
      >
        We found 2 results
      </caption>
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header"
            scope="col"
          >
            Address
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Postcode
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Property type
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Property reference
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row govuk-table__row--clickable"
        >
          <td
            class="govuk-table__cell"
          >
            16 Pitcairn House  St Thomass Square
          </td>
          <td
            class="govuk-table__cell"
          >
            E9 6PT
          </td>
          <td
            class="govuk-table__cell"
          >
            Dwelling
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              href="/properties/00023404"
            >
              00023404
            </a>
          </td>
        </tr>
        <tr
          class="govuk-table__row govuk-table__row--clickable"
        >
          <td
            class="govuk-table__cell"
          >
            1 Pitcairn House  St Thomass Square
          </td>
          <td
            class="govuk-table__cell"
          >
            E9 6PT
          </td>
          <td
            class="govuk-table__cell"
          >
            Dwelling
          </td>
          <td
            class="govuk-table__cell"
          >
            <a
              href="/properties/00023389"
            >
              00023389
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
